src/Pages/QuizCard.tsx

```diff
@@ -0,0 +1,105 @@
+// import { quizData } from "../QuizData/quizData";
+import { useEffect } from "react";
+import { useDataContext } from "../Context/QuizContextProvider";
+import { useNavigate, useParams } from "react-router-dom";
+import {
+  ScaleFade,
+  Button,
+  Flex,
+  Box,
+  Center,
+  Text,
+  Heading
+} from "@chakra-ui/react";
+import { useState } from "react";
+
+
+export type UserAnswer = {
+  questionId: number;
+  answer: string;
+};
+
+export function QuizCard() {
+  const {
+
+    userAnswer,
+    setUserAnswer,
+    quizData
+  } = useDataContext();
+  const navigate = useNavigate();
+  
+  const { quizId } = useParams();
+  const quizid = parseInt(quizId,10)
+  
+  const [questionIdx, setQuestionIdx] = useState<number>(0)
+  
+  const quizLength = quizData.data[0].questions.length + 1;
+  
+  useEffect(() => {
+    if (questionIdx === quizData.data[quizid].questions.length) {
+      navigate(`/dashboard/${quizid}`, { state: userAnswer });
+    }
+  }, [questionIdx]);
+
+  function answerManager(questionId: number, optionId: string) {
+    if (
+      quizLength !== quizData.data[quizid].questions[questionIdx].questionId
+    ) {
+      setUserAnswer((answers) => [
+        ...answers,
+        { questionId: questionId, answer: optionId }
+      ]);
+      setQuestionIdx(questionIdx + 1);
+    }
+  }
+
+  return (
+    <Box h="100vh">
+      <ScaleFade
+        in={true
+        }
+      >
+        <Center mt="2rem">
+          <Box borderRadius="10px" w="80%" className="glass-effect" p="1.5rem">
```

Comment:
this can be made consistent without mixing px, rem and % . 1 unit can be used for consistency